minio/minio
# avoid using background goroutine for caching http ranges

## Description

For some reason, caching HTTP ranges was implemented to download the data range in parallel in two separate goroutines:
- One that downloads the data range from the backend and serves it to the S3 client, and
- A second one that downloads the data from the backend and writes it to the cache mount.

I contrast, downloading a complete file uses a TeeReader in a single goroutine. It downloads the data range once from the backend and writes it to the cache mount while serving it to the S3 client.

This change modifies the cache code to uses the TeeReader in a single goroutine for downloading HTTP ranges.

The additional background goroutine remains for the case when `MINIO_CACHE_RANGE=off`. I suspect this case was the main motivation for the additional goroutine as the complete file should be downloaded to the cache, while only a range of it served to the S3 client.

## Motivation and Context

We have trouble with caching a larger file with our MinIO gateway for Tardigrade. The two goroutines are reading from the same Reader in parallel. The one serving the data range to the S3 client finishes first and closes the Reader. This causes the goroutine that is downloading the data range to the cache to fail. 

This failure is caught at: https://github.com/minio/minio/blob/5cc23ae0526d28874f25b100c1477b476ec39e26/cmd/disk-cache-backend.go#L715
This forces the removal of all data for this file from the cache.

As a result, our gateway:
- Doesn't have a working cache for larger files.
- Wastes more egress traffic from the backend than required.

## How to test this PR?

Use AWS CLI to download a larger file (tens or hundreds of MB) several times. Watch if the cache works as expected.

Cache settings:
```
MINIO_CACHE=on
MINIO_CACHE_DRIVES=/var/cache/storj
MINIO_CACHE_EXPIRY=180
MINIO_CACHE_QUOTA=80
MINIO_CACHE_WATERMARK_LOW=80
MINIO_CACHE_WATERMARK_HIGH=95
MINIO_CACHE_AFTER=0
```


## Types of changes

## Checklist:
